graphite.go

```diff
@@ -28,21 +31,32 @@ func NewClient(Host string, Port int, Prefix string, Protocol string) *Client {
 	}
 }
 
-// SendData - creates new connection to Graphite server and pushes batch of metrics in this single connection. Default connect timeout is set to 3s.
+// SendData - creates new connection to Graphite server and
+// pushes provided batch of metrics in this single connection.
+// Default connect timeout is set to 3s.
+// Returns error in case of problems establishing or closing the connection
+// but no error in case of problems sending data trough the connection
+// (which should not be a problem with such short-lived connections).
 //
-// SendData receives as argument map[string]int64 where string is metric name, float64 is metric value, example:
+// SendData receives as argument map[string]int64 where string is metric name,
+// float64 is metric value, example:
 //   map[string]float64{"test": 1234.1234, "test": 1234.1234}
 func (g *Client) SendData(data map[string]float64) error {
-	dataToSent := g.prepareData(data)
-	conn, err := net.DialTimeout(g.protocol, g.host+":"+strconv.Itoa(g.port), time.Second*3)
+	if g.timeOut == 0 {
```

Comment:
This particular part of the code makes it thread-unsafe. I don't know if it makes sense to make this code thread-safe intentionally, but it was that way before my changes.

I think we should either remove this or allow setting the timeout once in the constructor (to make sure it won't change in process of the SendData call). What do you think on that?

Reply:
I made code thread-safe again, no behavior changed but this fact is clarified in comment, as well as storing the default timeout value explicitly in the `Client`.